### PR TITLE
migrate kubernetes/node-feature-discovery jobs to eks cluster

### PR DESCRIPTION
**Pull request by iftachk**
jobs: migrate kubernetes/node-feature-discovery jobs to eks cluster

- Add missing resource blocks
/priority important-longterm
/area jobs

Part of https://github.com/kubernetes/test-infra/issues/29722

### CODE REVIEW

**Review by marquiz — COMMENTED** [review]
Thanks @iftachk for the patch.

Is there a reason to only convert some of the jobs and not all of them?

We also need to think about the resource requirements. Some of the jobs might want more than one cpu.

/cc @ArangoGutierrez

**Comment by iftachk** [comment]
@marquiz Thank you for your feedback

Regarding the per-resource definitions, feel free to review the PR and let me know which jobs might need more resources.

For the reason why I only changed some jobs and not all - please review this [issue|29722](https://github.com/kubernetes/test-infra/issues/29722) which includes some specific details on which jobs needs to be migrated eg ([kubernetes-sigs/node-feature-discovery](https://github.com/kubernetes-sigs/node-feature-discovery) | [Prow Results](https://prow.k8s.io/?repo=kubernetes-sigs%2Fnode-feature-discovery&cluster=default) | [Search Results](https://cs.k8s.io/?q=pull-node-feature-discovery-build-image-cross-generic%7Cpull-node-feature-discovery-build-image-generic%7Cpull-node-feature-discovery-build-master%7Cpull-node-feature-discovery-verify-master%7Cpull-node-feature-discovery-build-gh-pages-generic%7Cpull-node-feature-discovery-verify-docs-master%7Cpostsubmit-node-feature-discovery-e2e-test%7Cpostsubmit-node-feature-discovery-verify-master&i=nope&files=config%2Fjobs%2Fkubernetes-sigs%2Fnode-feature-discovery))

For any follow-up questions, I would recommend reaching out to sig-k8s-infra or sig-testing

**Review by rjsadow — COMMENTED** [review]
There are a few jobs listed here that we wont be able to migrate quite yet.

**Comment by ArangoGutierrez** [comment]
> Thanks @iftachk for the patch.
> 
> Is there a reason to only convert some of the jobs and not all of them?
> 
> We also need to think about the resource requirements. Some of the jobs might want more than one cpu.
> 
> /cc @ArangoGutierrez

Agree @iftachk NFD could crash with just 1 CPU, for the rest it looks ok to me

**Comment by iftachk** [comment]
@ArangoGutierrez, as requested I've updated the resource blocks for the migrated NFD jobs
Would appreciate your feedback and approval in case it looks good to you.
Thanks,

**Comment by ArangoGutierrez** [comment]
> @ArangoGutierrez, as requested I've updated the resource blocks for the migrated NFD jobs Would appreciate your feedback and approval in case it looks good to you. Thanks,

going to test the full e2e run with the given resources and reply back shortly after

File: config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-presubmits-generic.yaml [hunk 1]
```diff
@@ -61,7 +70,15 @@ presubmits:
         - runner.sh
         args:
         - scripts/test-infra/build-image-cross.sh
+        resources:
+          limits:
+            cpu: 2
+            memory: 4Gi
```
Comment (marquiz) [inline config/jobs/kubernetes-sigs/node-feature-discovery/node-feature-discovery-presubmits-generic.yaml:76]: Oh, and I think 4gi is too little for this

```suggestion
            memory: 8Gi
```

**Pull request merged**